test/base/src/androidTest/kotlin/io/realm/test/shared/QueryTests.kt

```diff
@@ -0,0 +1,561 @@
+ * limitations under the License.
+ */
+
+package io.realm.test.shared
+
+import io.realm.QuerySort
+import io.realm.Realm
+import io.realm.RealmConfiguration
+import io.realm.entities.Sample
+import io.realm.find
+import io.realm.test.platform.PlatformUtils
+import kotlin.test.AfterTest
+import kotlin.test.BeforeTest
+import kotlin.test.Ignore
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertFailsWith
+import kotlin.test.assertTrue
+
+@Suppress("LargeClass")
+class QueryTests {
+
+    private lateinit var tmpDir: String
+    private lateinit var realm: Realm
+
+    @BeforeTest
+    fun setup() {
+        tmpDir = PlatformUtils.createTempDir()
+        val configuration = RealmConfiguration.Builder(schema = setOf(Sample::class))
+            .path("$tmpDir/default.realm").build()
+        realm = Realm.open(configuration)
+    }
+
+    @AfterTest
+    fun tearDown() {
+        if (!realm.isClosed()) {
+            realm.close()
+        }
+        PlatformUtils.deleteTempDir(tmpDir)
+    }
+
+    @Test
+    fun find() {
+        realm.query(Sample::class)
+            .find { results -> assertEquals(0, results.size) }
+
+        val stringValue = "some string"
+        realm.writeBlocking {
+            // Queries inside a write transaction are live and can be reused within the closure
+            val query = query(Sample::class)
+
+            assertEquals(0, query.find().size)
+            copyToRealm(Sample().apply { stringField = stringValue })
+            assertEquals(1, query.find().size)
+        }
+
+        // No filter
+        realm.query(Sample::class)
+            .find { results -> assertEquals(1, results.size) }
+
+        // Filter by string
+        realm.query(Sample::class, "stringField = $0", stringValue)
+            .find { results -> assertEquals(1, results.size) }
+
+        // Filter by string that doesn't match
+        realm.query(Sample::class, "stringField = $0", "invalid string")
+            .find { results -> assertEquals(0, results.size) }
+    }
+
+    @Test
+    fun find_malformedQueryThrows() {
+        assertFailsWith<IllegalArgumentException> {
+            realm.query(Sample::class, "stringField = $0")
+        }.let {
+            assertTrue(it.message!!.contains("Have you specified all parameters"))
+        }
+
+        assertFailsWith<IllegalArgumentException> {
+            realm.query(Sample::class, "stringField = 42")
+        }.let {
+            assertTrue(it.message!!.contains("Wrong query field"))
+        }
+
+        assertFailsWith<IllegalArgumentException> {
+            realm.query(Sample::class, "nonExistingField = 13")
+        }.let {
+            assertTrue(it.message!!.contains("Wrong query field"))
+        }
+    }
+
+    @Test
+    @Ignore
+    fun asFlow() {
+        // TODO
+    }
+
+    @Test
+    @Ignore
+    fun asFlow_throwsInsideWrite() {
+        // TODO
+    }
+
+    @Test
+    fun composedQuery() {
+        val value1 = 1
```

Comment:
These doesn't really seem needed as you only use them once when creating the data?